### PR TITLE
Fixes close event for ledger

### LINKED ISSUE #11688: Websites are not appearing in the ledger table when you close tab
### Description

Sometimes websites are not being added into the ledger even though they've met the conditions outlined in the "Advanced Settings". This doesn't always happen, but I've managed to reproduce it a few times.

### Steps to Reproduce

*Note:* Steps are not 100% and might require a few tries before you see the issue appearing

  1. launch `v0.19.67` rev: b3a897eef1f1d8d6594e2d8e862bf7a874e86e73 
  2. enable "Payments" via `about:preferences#payments`
  3. once the wallet is created, open a new tab and visit a website (I used boardgamegeek.com)
  4. wait for about a minute and switch back to the `about:preferences#payments` tab
  5. you'll notice that sometimes the websites are not appearing under the table

**Actual result:**

Websites are not being added into the ledger table once they've met the conditions outlined in the wallet via "Advanced Settings".

Example GIF: https://drive.google.com/file/d/0B9irDS_rLz-lX1VaajFGT2w5d0U/view?usp=sharing

**Expected result:**

Websites should always be added into the ledger table once they've met the conditions outlined in the wallet via "Advanced Settings", unless they're on the "exclude" list.

**Reproduces how often:** [What percentage of the time does it reproduce?]

Using the above STR/the GIF example... I would say there's about a 30% chance of reproducing the issue.

### Brave Version

**about:brave info:**

> Brave: 0.19.67 
> rev: b3a897eef1f1d8d6594e2d8e862bf7a874e86e73 
> Muon: 4.5.9 
> libchromiumcontent: 62.0.3202.62 

**Reproducible on current live release:**

Yes, I've ran into similar issues before where websites never appear in the ledger table.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #11688

Auditors:

Test Plan:
- enable payments
- open new tab
- visit site in this tab and stay on this site for 15s
- close tab
- site should appear in ledger table

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/common/state/pageDataState.js [hunk 1]
```diff
@@ -52,6 +52,28 @@ const pageDataState = {
     return state.setIn(['pageData', 'last', 'tabId'], tabId)
   },
 
+  saveLastClosedTab: (state, tabValue) => {
+    if (tabValue == null) {
+      return state
+    }
+
+    return state.setIn(['pageData', 'last', 'closedTabValue'], tabValue)
```
Comment (bbondy) [inline app/common/state/pageDataState.js]: I think this is going to cause a privacy problem for incognito mode

Comment (NejcZdovc) [reply app/common/state/pageDataState.js]: fixed

**Comment by bbondy** [comment]
master: 603182688b687041c19e96b705f3e7565d174c66
0.21.x: e8eb32d29e8bec65496a200aa2442b312b78adee
0.20.x: 95f509ce5c45bbe383b457a9b372eb0d29ce3b9d
0.19.x: 4693896a7cce0630b2cedf710135f9d47486ad56

**Comment by bbondy** [comment]
merged for hotfix 2

**Pull request merged**